File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
            
            
        int i=m+n-1;
        int j=n-1;
        int k=m-1;
        while(j>=0 and k>=0){
            
                if(nums1[k]>nums2[j]){
                        
                            nums1[i]=nums1[k];
                            k--;
                }
                else{
                            
                            nums1[i]=nums2[j];
                            j--;
                }
            
                i--;
            
        }
        
        cout<<j<<k<<endl;
        while(j>=0){
            
                    nums1[i]=nums2[j];
                    j--;
                    i--;
            
        }
        
        while(k>=0){
            
                    nums1[i]=nums1[k];
                    k--;
                    i--;
            
        }
        
        
        
        
        
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        
        """
        Do not return anything, modify nums1 in-place instead.
        """
        
        
        i=len(nums1)-len(nums2)-1
        j=len(nums2)-1
        k=len(nums1)-1
        while(i>=0 and j>=0):
               
            
                if nums1[i]<nums2[j]:
                        nums1[k]=nums2[j]
                        k-=1
                        j-=1
                        
                elif nums1[i]>=nums2[j]:
                        nums1[k]=nums1[i]
                        i-=1
                        k-=1
                
        
        
        while (j<0 and i>=0):
                    nums1[k]=nums1[i]
                    k-=1
                    i-=1
                    
        while( i<0 and j>=0):
                    
                    
                    nums1[k]=nums2[j]
                    k-=1
                    j-=1
                        
                        
                        
                        
                        
                        
                        
                    
                    
                    
                
            
            
            
            
            
            
        
        